pkg/quota/v1/evaluator/core/services.go

```diff
@@ -120,22 +122,39 @@ func (p *serviceEvaluator) Usage(item runtime.Object) (corev1.ResourceList, erro
 	// default service usage
 	result[serviceObjectCountName] = *(resource.NewQuantity(1, resource.DecimalSI))
 	result[corev1.ResourceServices] = *(resource.NewQuantity(1, resource.DecimalSI))
-	result[corev1.ResourceServicesLoadBalancers] = resource.Quantity{Format: resource.DecimalSI}
-	result[corev1.ResourceServicesNodePorts] = resource.Quantity{Format: resource.DecimalSI}
+	// check nodeport and load balancer only if the runtime svc is one of them.
 	switch svc.Spec.Type {
 	case corev1.ServiceTypeNodePort:
 		// node port services need to count node ports
 		value := resource.NewQuantity(int64(ports), resource.DecimalSI)
 		result[corev1.ResourceServicesNodePorts] = *value
 	case corev1.ServiceTypeLoadBalancer:
-		// load balancer services need to count node ports and load balancers
-		value := resource.NewQuantity(int64(ports), resource.DecimalSI)
-		result[corev1.ResourceServicesNodePorts] = *value
+		// load balancer services need to count node ports. If creation of node ports
+		// is suppressed only ports with explicit NodePort values are counted.
+		// nodeports won't be allocated yet, so we can't simply count the actual values.
+		// We need to look at the intent.
+		if feature.DefaultFeatureGate.Enabled(features.ServiceLBNodePortControl) &&
+			*svc.Spec.AllocateLoadBalancerNodePorts == false {
```

Comment:
make sure `svc.Spec.AllocateLoadBalancerNodePorts` is not nil here. This code also runs in the controller manager, so it is possible for the feature gate to be enabled in the controller manager and not in the API server